app/src/utils/data/contributions.ts

```diff
@@ -368,3 +375,103 @@ export function grantDonationListener(
     off: () => grantRoundManager.value.off(filter, listener),
   };
 }
+
+/**
+ * @notice given a userAddress and the contributions, return only the contributions for that address
+ * @param userAddress
+ * @param contributions
+ */
+export function filterContributionsByUserAddress(userAddress: string, contributions: Contribution[]) {
+  if (contributions && userAddress) {
+    return contributions.filter((contribution: Contribution) => {
+      return contribution.address === userAddress;
+    });
+  }
+  return undefined;
+}
+
+/**
+ * @notice filters through the grantrounds and returns the necessary attributes for  contribution detail screens
+ * @param grantRounds
+ * @param grantId
+ * @param grantRoundMeta
+ */
+function filterGrantRoundsForContributions(
+  grantRounds: GrantRound[],
+  grantId: number,
+  grantRoundMeta?: Record<string, GrantRoundMetadata>
+) {
+  let roundName = '...';
+
+  if (grantRoundMeta) {
+    grantRounds.find((grantRound) => {
+      const metadata = grantRoundMeta[metadataId(grantRound.metaPtr)];
+      if (metadata && metadata.grants?.includes(grantId)) {
+        roundName = metadata.name;
+      }
+      return roundName;
+    });
+  }
+
+  return roundName;
+}
+
+/***
+ * @notice
+ * @param userAddress
+ * @param contributions
+ * @param grants
+ * @param grantMetaData
+ * @param grantRounds
+ * @param grantRoundsMetaData
+ */
+export function filterContributionGrantData(
+  userAddress: string,
+  contributions: Contribution[],
+  grants: Grant[],
+  grantRounds: GrantRound[],
+  grantMetaData?: Record<string, GrantMetadata>,
+  grantRoundsMetaData?: Record<string, GrantRoundMetadata>
+): ContributionDetail[] {
+  if (!userAddress || !contributions) {
```

Comment:
Same comment on checking array length. There might be a few others like this that I missed so maybe just give that a quick review everywhere